code/modules/mob/living/carbon/human/human_stripping.dm

```diff
@@ -0,0 +1,261 @@
+	user.log_message(log_message, LOG_ATTACK, color="red", log_globally=FALSE)
+	item.add_fingerprint(src)
+
+	var/result = start_unequip_mob(item, source, user, POCKET_STRIP_DELAY)
+
+	if(!result)
+		warn_owner(source)
+
+	return result
+
+/datum/strippable_item/mob_item_slot/needs_jumpsuit/pocket/proc/warn_owner(atom/owner)
+	to_chat(owner, "<span class='warning'>You feel your [pocket_side] pocket being fumbled with!</span>")
+
+/datum/strippable_item/mob_item_slot/needs_jumpsuit/pocket/left
+	key = STRIPPABLE_ITEM_LPOCKET
+	item_slot = ITEM_SLOT_LPOCKET
+	pocket_side = "left"
+
+/datum/strippable_item/mob_item_slot/needs_jumpsuit/pocket/right
+	key = STRIPPABLE_ITEM_RPOCKET
+	item_slot = ITEM_SLOT_RPOCKET
+	pocket_side = "right"
+
+/proc/get_strippable_alternate_action_internals(obj/item/item, atom/source)
+	if(!iscarbon(source))
+		return
+
+	var/mob/living/carbon/carbon_source = source
+
+	var/obj/item/clothing/mask = carbon_source.wear_mask
+	if(!istype(mask))
+		return
+
+	if((mask.clothing_flags & MASKINTERNALS) && istype(item, /obj/item/tank))
+		return isnull(carbon_source.internal) ? "enable_internals" : "disable_internals"
+
+/proc/strippable_alternate_action_internals(obj/item/item, atom/source, mob/user)
+	var/obj/item/tank/tank = item
+	if(!istype(tank))
+		return
+
+	var/mob/living/carbon/carbon_source = source
+	if(!istype(carbon_source))
+		return
+
+	var/obj/item/clothing/mask = carbon_source.wear_mask
+	if(!istype(mask) || !(mask.clothing_flags & MASKINTERNALS))
+		return
+
+	carbon_source.visible_message(
+		"<span class='danger'>[user] tries to [isnull(carbon_source.internal) ? "open": "close"] the valve on [source]'s [item.name].</span>",
+		"<span class='userdanger'>[user] tries to [isnull(carbon_source.internal) ? "open": "close"] the valve on your [item.name].</span>",
+		ignored_mobs = user,
+	)
+
+	to_chat(user, "<span class='notice'>You try to [isnull(carbon_source.internal) ? "open": "close"] the valve on [source]'s [item.name]...</span>")
+
+	if(!do_mob(user, carbon_source, INTERNALS_TOGGLE_DELAY))
+		return
+
+	if(carbon_source.internal)
+		carbon_source.internal = null
+
+		// This isn't meant to be FALSE, it correlates to the icon's name.
+		carbon_source.update_internals_hud_icon(0)
+	else if(!QDELETED(item))
+		if((carbon_source.wear_mask?.clothing_flags & MASKINTERNALS) || carbon_source.getorganslot(ORGAN_SLOT_BREATHING_TUBE))
+			carbon_source.internal = item
+			carbon_source.update_internals_hud_icon(1)
+
+	carbon_source.visible_message(
+		"<span class='danger'>[user] [isnull(carbon_source.internal) ? "closes": "opens"] the valve on [source]'s [item.name].</span>",
+		"<span class='userdanger'>[user] [isnull(carbon_source.internal) ? "closes": "opens"] the valve on your [item.name].</span>",
+		ignored_mobs = user,
+	)
+
+	to_chat(user, "<span class='notice'>You [isnull(carbon_source.internal) ? "close" : "open"] the valve on [source]'s [item.name].</span>")
```

Comment:
Do these have to be global procs?

Reply:
As far as I know, absolutely not... But I suspect there's a reason why they were made global, this is taken from tg. Maybe performance improvements, maybe because there's no reason to put them on any one datum.

I can move them onto some type if you'd like, but I don't know the potential implications of it.